models/comments.js

```diff
@@ -0,0 +1,68 @@
+const db = require("../db/connection");
+
+exports.selectCommentsById = (article_id) => {
+  return db
+    .query(
+      `SELECT comment_id, votes, created_at, author, body, article_id 
+       FROM comments 
+       WHERE article_id = $1 
+       ORDER BY created_at DESC`,
+      [article_id]
+    )
+    .then((result) => {
+      return result.rows;
+    })
+    .catch((err) => {
+      throw err;
+    });
+};
+
+exports.insertComment = (article_id, username, body) => {
+  return db
+    .query(
+      `INSERT INTO comments (article_id, author, body, votes, created_at)
+       VALUES ($1, $2, $3, 0, NOW())
+       RETURNING *;`,
+      [article_id, username, body]
+    )
+    .then((result) => {
+      return result.rows[0];
+    })
+    .catch((err) => {
+      throw err;
+    });
+};
+
+exports.checkArticleExists = (article_id) => {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Article not found" });
+      }
+      return true;
+    });
+};
+
+exports.checkUserExists = (username) => {
+  return db
+    .query(`SELECT * FROM users WHERE username = $1`, [username])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "User not found" });
+      }
+      return true;
+    });
+};
+
+exports.removeCommentById = (comment_id) => {
+  return db
+    .query("DELETE FROM comments WHERE comment_id = $1 RETURNING *;", [
+      comment_id,
+    ])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Comment not found" });
```

Comment:
Don't doe destructure. --> result
if(result.rowCount === 0) --> error

Reply:
I've now updated to rowCount thanks for the advice